x/accounts/vesting/vesting.go

```diff
@@ -0,0 +1,493 @@
+		}
+		delFreeAmt, err := bva.DelegatedFree.Get(ctx, coin.Denom)
+		if err != nil {
+			return err
+		}
+		delVestingAmt, err := bva.DelegatedVesting.Get(ctx, coin.Denom)
+		if err != nil {
+			return err
+		}
+
+		// compute x and y per the specification, where:
+		// X := min(DF, D)
+		// Y := min(DV, D - X)
+		x := math.MinInt(delFreeAmt, coin.Amount)
+		y := math.MinInt(delVestingAmt, coin.Amount.Sub(x))
+
+		delVestingCoin := sdk.NewCoin(coin.Denom, delVestingAmt)
+		delFreeCoin := sdk.NewCoin(coin.Denom, delFreeAmt)
+		if !x.IsZero() {
+			xCoin := sdk.NewCoin(coin.Denom, x)
+			newDelFree := delFreeCoin.Sub(xCoin)
+			err = bva.DelegatedVesting.Set(ctx, newDelFree.Denom, newDelFree.Amount)
+			if err != nil {
+				return err
+			}
+		}
+
+		if !y.IsZero() {
+			yCoin := sdk.NewCoin(coin.Denom, y)
+			newDelVesting := delVestingCoin.Sub(yCoin)
+			err = bva.DelegatedVesting.Set(ctx, newDelVesting.Denom, newDelVesting.Amount)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (bva BaseVesting) getBalance(ctx context.Context, sender, denom string) (*sdk.Coin, error) {
+	// Query account balance for the sent denom
+	balanceQueryReq := banktypes.NewQueryBalanceRequest(sdk.AccAddress(sender), denom)
+	resp, err := accountstd.QueryModule[banktypes.QueryBalanceResponse](ctx, balanceQueryReq)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Balance, nil
+}
+
+func (bva BaseVesting) checkTokensSendable(ctx context.Context, sender string, amount, vestingCoins sdk.Coins) error {
+	// Get locked token
+	lockedCoins, err := bva.LockedCoinsFromVesting(ctx, vestingCoins)
+	if err != nil {
+		return err
+	}
+
+	// Check if any sent tokens is exceeds vesting account balances
+	for _, coin := range amount {
+		balance, err := bva.getBalance(ctx, sender, coin.Denom)
+		if err != nil {
+			return err
+		}
+		locked := sdk.NewCoin(coin.Denom, lockedCoins.AmountOf(coin.Denom))
+
+		spendable, hasNeg := sdk.Coins{*balance}.SafeSub(locked)
+		if hasNeg {
+			return errorsmod.Wrapf(sdkerrors.ErrInsufficientFunds,
+				"locked amount exceeds account balance funds: %s > %s", locked, balance)
+		}
+
+		if _, hasNeg := spendable.SafeSub(coin); hasNeg {
+			if len(spendable) == 0 {
+				spendable = sdk.Coins{sdk.NewCoin(coin.Denom, math.ZeroInt())}
+			}
+			return errorsmod.Wrapf(
+				sdkerrors.ErrInsufficientFunds,
+				"spendable balance %s is smaller than %s",
+				spendable, coin,
+			)
+		}
+	}
+
+	return nil
+}
+
+// this function only a placeholder so that there will be no routing err return
+// all bundler payment messages will be ignore, this is to prevent bypass vesting
+// account check for send, delegate and undelegate action when execute messages
+// by pass it in bundler. Since vesting doesn't have a handler for payBundler message,
+// this could lead to accounts keeper executes the messages directly without going
+// through vesting account ExecuteMessages handler.
```

Comment:
This comment is really confusing.

```suggestion
// payBundler is only a placeholder so that there will be no routing err returned.
// All bundler payment messages will be ignored. This is to prevent bypassing the vesting
// account checks for the send, delegate and undelegate actions when executing messages
// by passing them in a bundler. Since vesting doesn't have a handler for payBundler message,
// this could lead to the accounts keeper executing the messages directly without going
// through vesting account ExecuteMessages handler.
```

Reply:
hmmm i think this comment along the func will be remove, since the paybundler is no longer exists (or not yet implemented in the current version of account abstraction) so this func is not needed anymore